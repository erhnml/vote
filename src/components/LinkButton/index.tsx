import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';

interface LinkButtonProps {
  to: string,
  icon?: React.ReactNode,
  title: string,
}

const LinkButton = ({to, icon, title}:LinkButtonProps) => {
  return (
    <StyledLink to={to}>
      { icon }
      <Title icon={icon}>{title}</Title>
    </StyledLink>
  );
}

const StyledLink = styled(Link)`
  display: flex;
  align-items: center;
  padding: 4px 15px;
  color: black;
  text-decoration: none;
  background-color: ${({ theme }) => theme.colors.lightGray};
  font-size: 14px;
  border-radius: 3px;
  height: 38px;
`;

const Title = styled.span`
    ${({ icon }:{icon: React.ReactNode}) => icon && `
    margin-left: 10px;
  `}
`;

export default LinkButton;
