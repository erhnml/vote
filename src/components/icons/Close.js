import * as React from "react";

function SvgClose(props) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      width={24}
      height={24}
      viewBox="0 0 612 612"
      {...props}
    >
      <path
        d="M415.338 196.662c-7.535-7.535-19.737-7.535-27.253 0l-82.181 82.18-81.033-81.032c-7.478-7.478-19.584-7.478-27.042 0-7.478 7.478-7.478 19.584 0 27.042l81.033 81.033-81.587 81.587c-7.535 7.535-7.535 19.736 0 27.253s19.737 7.517 27.253 0l81.588-81.587 81.032 81.032c7.478 7.478 19.584 7.478 27.043 0 7.478-7.478 7.478-19.584 0-27.043l-81.033-81.032 82.181-82.18c7.534-7.516 7.534-19.736-.001-27.253zM306 0C136.992 0 0 136.992 0 306s136.992 306 306 306c168.988 0 306-137.012 306-306S475.008 0 306 0zm0 573.75C158.125 573.75 38.25 453.875 38.25 306S158.125 38.25 306 38.25 573.75 158.125 573.75 306 453.875 573.75 306 573.75z"
        fill="#fff"
        data-original="#000000"
        xmlns="http://www.w3.org/2000/svg"
      />
    </svg>
  );
}

export default SvgClose;
