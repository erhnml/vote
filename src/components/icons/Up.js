import * as React from "react";

function SvgUp(props) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      width={24}
      height={24}
      viewBox="0 0 123.969 123.97"
      {...props}
    >
      <path d="M120.266 95.584c5.301-5.801 4.801-14.801-.8-20.301l-47.3-47.4c-2.8-2.8-6.5-4.2-10.2-4.2s-7.399 1.4-10.2 4.2l-47.299 47.3c-5.5 5.5-6.1 14.6-.8 20.3 5.6 6.101 15.099 6.3 20.9.5l30.3-30.3a9.959 9.959 0 0114.1 0l30.3 30.3c5.9 5.9 15.4 5.701 20.999-.399z" />
    </svg>
  );
}

export default SvgUp;
