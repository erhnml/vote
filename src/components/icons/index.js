export { default as Down } from "./Down";
export { default as Up } from "./Up"
export { default as LeftArrow } from "./LeftArrow";
export { default as Close } from "./Close";
export { default as Delete } from "./Delete";
export { default as Plus } from "./Add";