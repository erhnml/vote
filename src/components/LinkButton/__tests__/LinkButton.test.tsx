
import React from 'react';
import {  mount } from 'enzyme';
import {  MemoryRouter } from "react-router-dom";
import LinkButton from '../index';
import { Delete } from '../../icons';

describe('LinkButton', () => {
  it('renders correctly', () => {
    expect(
      <MemoryRouter>
        <LinkButton to="/" title="Return to List" />
      </MemoryRouter>
    ).toMatchSnapshot();
  });

  it('mount correctly', () => {
    expect(() => mount(
      <MemoryRouter>
        <LinkButton to="/" title="Return to List" />
      </MemoryRouter>
    )).not.toThrow();
  });

  it('should render with icon', () => {
    const wrapper = mount(
      <MemoryRouter>
        <LinkButton 
          to="/"
          title="Return to List"
          icon={<Delete />} 
        />
      </MemoryRouter>
      );
    expect(wrapper.find(Delete)).toHaveLength(1);
  });
})
