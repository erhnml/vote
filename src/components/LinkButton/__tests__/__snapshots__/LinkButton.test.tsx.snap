// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButton renders correctly 1`] = `
<MemoryRouter>
  <LinkButton
    title="Return to List"
    to="/"
  />
</MemoryRouter>
`;
